include/boost/multiprecision/cpp_int/misc.hpp

```diff
@@ -208,10 +211,14 @@ eval_convert_to(R* result, const cpp_int_backend<MinBits1, MaxBits1, SignType1,
          if (bits_to_keep < (std::ptrdiff_t)cpp_int_backend<MinBits1, MaxBits1, SignType1, Checked1, Allocator1>::limb_bits)
          {
             if(index != backend.size() - 1)
-               mask <<= cpp_int_backend<MinBits1, MaxBits1, SignType1, Checked1, Allocator1>::limb_bits - bits_to_keep;
+            {
+               const std::ptrdiff_t left_shift_amount = static_cast<std::ptrdiff_t>(static_cast<std::ptrdiff_t>(cpp_int_backend<MinBits1, MaxBits1, SignType1, Checked1, Allocator1>::limb_bits) - bits_to_keep);
+
+               mask = static_cast<limb_type>(mask << left_shift_amount);
```

Comment:
Can we really not use <<= here?  It's fine, but it's just harder to read...

Reply:
> Can we really not use <<= here? It's fine, but it's just harder to read...

Alas the verbosity of my lines is scaling with my increasing age. Sure. I can terse-ify.